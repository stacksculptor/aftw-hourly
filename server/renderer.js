const path = require('path');
const fs = require('fs');
const _ = require('lodash');
const { renderApp } = require('./importer');

const buildPath = path.resolve(__dirname, '..', 'build');

// The HTML build file is generated from the `public/index.html` file
// and used as a template for server side rendering. The application
// head and body are injected to the template from the results of
// calling the `renderApp` function imported from the bundle above.
const indexHtml = fs.readFileSync(path.join(buildPath, 'index.html'), 'utf-8');

const reNoMatch = /($^)/;
const template = _.template(indexHtml, {
  // Interpolate variables in the HTML template with the following
  // syntax: <!--!variableName-->
  //
  // This syntax is very intentional: it works as a HTML comment and
  // doesn't render anything visual in the dev mode, and in the
  // production mode, HtmlWebpackPlugin strips out comments using
  // HTMLMinifier except those that aren't explicitly marked as custom
  // comments. By default, custom comments are those that begin with a
  // ! character.
  //
  // Note that the variables are _not_ escaped since we only inject
  // HTML content.
  //
  // See:
  // - https://github.com/ampedandwired/html-webpack-plugin
  // - https://github.com/kangax/html-minifier
  // - Plugin options in the production Webpack configuration file
  interpolate: /<!--!([\s\S]+?)-->/g,
  // Disable evaluated and escaped variables in the template
  evaluate: reNoMatch,
  escape: reNoMatch,
});

exports.render = function(requestUrl, context, preloadedState) {
  const { head, body } = renderApp(requestUrl, context, preloadedState);

  // Preloaded state needs to be passed for client side too.
  // For security reasons we ensure that preloaded state is considered as a string
  // by replacing '<' character with its unicode equivalent.
  // http://redux.js.org/docs/recipes/ServerRendering.html#security-considerations
  const serializedState = JSON.stringify(preloadedState).replace(/</g, '\\u003c');
  const preloadedStateScript = `
      <script>window.__PRELOADED_STATE__ = ${serializedState};</script>
  `;

  return template({ title: head.title.toString(), preloadedStateScript, body });
};
