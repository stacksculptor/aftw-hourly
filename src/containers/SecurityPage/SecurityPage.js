import React from 'react';
import PropTypes from 'prop-types';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { isScrollingDisabled } from '../../ducks/UI.duck';
import { Page } from '../../components';
import { TopbarContainer } from '../../containers';

export const SecurityPageComponent = props => {
  const { authInfoError, logoutError, scrollingDisabled } = props;

  return (
    <Page
      authInfoError={authInfoError}
      logoutError={logoutError}
      title="Security"
      scrollingDisabled={scrollingDisabled}
    >
      <TopbarContainer />
    </Page>
  );
};

SecurityPageComponent.defaultProps = {
  authInfoError: null,
  logoutError: null,
};

const { bool, instanceOf } = PropTypes;

SecurityPageComponent.propTypes = {
  authInfoError: instanceOf(Error),
  logoutError: instanceOf(Error),
  scrollingDisabled: bool.isRequired,
};

const mapStateToProps = state => {
  // Page needs authInfoError and logoutError
  const { authInfoError, logoutError } = state.Auth;
  return {
    authInfoError,
    logoutError,
    scrollingDisabled: isScrollingDisabled(state),
  };
};

const SecurityPage = compose(connect(mapStateToProps))(SecurityPageComponent);

export default SecurityPage;
