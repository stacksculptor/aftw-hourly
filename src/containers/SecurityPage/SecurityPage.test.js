import React from 'react';
import { BrowserRouter } from 'react-router';
import renderer from 'react-test-renderer';
import SecurityPage from './SecurityPage';

describe('SecurityPage', () => {
  it('matches snapshot', () => {
    const component = renderer.create(
      <BrowserRouter>
        <SecurityPage />
      </BrowserRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
