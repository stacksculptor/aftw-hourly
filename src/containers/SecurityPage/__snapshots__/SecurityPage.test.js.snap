// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecurityPage matches snapshot 1`] = `
<Page
  authInfoError={null}
  logoutError={null}
  scrollingDisabled={false}
  title="Security"
>
  <withRouter(Connect(TopbarContainerComponent)) />
</Page>
`;
